src/App.tsx

```diff
@@ -15,81 +16,160 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-enum SortType {
-  NONE,
-  ALPABET,
-  LENGTH,
-}
-
-type ReorderOptions = {
-  sortType: SortType,
+type State = {
+  goods: string[],
   isReversed: boolean,
+  sortBy: string,
+  resetVisibility: boolean,
+  classVisibility: string,
+  isReverseClicked: boolean,
 };
 
-// Use this function in the render to prepare goods
-export function getReorderedGoods(
-  goods: string[],
-  { sortType, isReversed }: ReorderOptions,
-) {
-  // To avoid the original array mutation
-  const visibleGoods = [...goods];
+type ButtonsMap = {
+  ALPHABETICALLY: string,
+  BUY_LENGTH: string,
+};
+
+const buttonsMap: ButtonsMap = {
+  ALPHABETICALLY: 'ALPHABETICALLY',
+  BUY_LENGTH: 'BUY_LENGTH',
+};
 
-  // Sort and reverse goods if needed
-  // eslint-disable-next-line no-console
-  console.log(sortType, isReversed);
+export class App extends React.Component<{}, State> {
+  state = {
+    goods: goodsFromServer,
+    isReversed: false,
+    sortBy: '',
+    resetVisibility: false,
+    classVisibility: '',
+    isReverseClicked: false,
+  };
 
-  return visibleGoods;
-}
+  toShowButton = () => {
+    this.setState({ resetVisibility: true });
+  };
 
-// DON'T save goods to the state
-// type State = {
-//   isReversed: boolean,
-//   sortType: SortType,
-// };
-
-export const App: React.FC = () => {
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
-
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
-
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
+  reverse = () => {
+    this.setState(state => ({
+      isReversed: !state.isReversed,
+      resetVisibility: true,
+      isReverseClicked: !state.isReverseClicked,
```

Comment:
Why do you need both `isReversed` and `isReverseClicked`?
You can use only first one